### PR TITLE
MIM-169: Create offer - Update listing status when no applicants

### PR DESCRIPTION
**Pull request by momentiris**
https://linear.app/mimer-onecore/issue/MIM-169/lagg-till-listing-status-noapplicants-samt-closed

### CODE REVIEW

File: src/adapters/leasing-adapter.ts [hunk 1]
```diff
@@ -447,22 +448,27 @@ const getApplicantByContactCodeAndListingId = async (
...
-    )
-    return undefined
+    logger.error(error, 'Error fetching detailed applicant data by listing id')
+
+    if (!(error instanceof AxiosError)) {
+      return { ok: false, err: 'unknown' }
+    }
+
+    if (error.response?.status === 404) {
+      return { ok: false, err: 'not-found' }
```
Comment (driatic) [inline src/adapters/leasing-adapter.ts:468]: We should get the error-codes from types and not have them as strings.

File: src/services/lease-service/index.ts [hunk 2]
```diff
@@ -1149,12 +1149,24 @@ export const routes = (router: KoaRouter) => {
...
-      await leasingAdapter.getListingByIdWithDetailedApplicants(
-        ctx.params.listingId
-      )
+    const result = await leasingAdapter.getDetailedApplicantsByListingId(
+      Number(ctx.params.listingId)
+    )
 
-    ctx.body = { content: responseData, ...metadata }
+    if (!result.ok) {
+      if (result.err === 'not-found') {
```
Comment (driatic) [inline src/services/lease-service/index.ts:1157]: We had a discussion about passing down the status code from the adapter last week. Could we apply this here and just proxy everything?

**Review by driatic — CHANGES_REQUESTED** [review]
Overall looks great! Just some nagging about the error codes as per last weeks discussion :)

File: src/adapters/leasing-adapter.ts (hunk 1 above)
Comment (momentiris) [reply src/adapters/leasing-adapter.ts:468]: What about add those error codes to this endpoint when we need them? I did actually re-write the endpoint but adding error codes here and in core feels a bit out of scope.

File: src/services/lease-service/index.ts (hunk 2 above)
Comment (momentiris) [reply src/services/lease-service/index.ts:1157]: > What about add those error codes to this endpoint when we need them? I did actually re-write the endpoint but adding error codes here and in leasing feels a bit out of scope.


If you think this is valid then perhaps this is out of scope for this particular endpoint here too?

Comment (momentiris) [reply src/services/lease-service/index.ts:1157]: Did however use error codes on the adapter method i introduced
https://github.com/Bostads-AB-Mimer/onecore-core/pull/190/commits/83ee4c4d42e70d967b37170c2793a2189cc9ab61

**Pull request merged**